src/android/Sketch.java

```diff
@@ -265,44 +272,63 @@ public void run() {
 
     private void saveDrawing(Intent intent) {
         Bundle extras = intent.getExtras();
-        byte[] drawingData = null;
+        String drawingPath = null;
         String output = null;
 
-        if (extras != null &&
-                extras.containsKey(TouchDrawActivity.DRAWING_RESULT_PARCELABLE)) {
-            drawingData = extras.getByteArray(TouchDrawActivity.DRAWING_RESULT_PARCELABLE);
+        if (extras != null && extras.containsKey(TouchDrawActivity.DRAWING_RESULT_PARCELABLE)) {
+            drawingPath = extras.getString(TouchDrawActivity.DRAWING_RESULT_PARCELABLE);
+            LOG.d(TAG, "Signaled we have a temp file in: " + drawingPath);
         }
 
-        if (drawingData == null || drawingData.length == 0) {
+        // Error out if we didn't get back a file path
+        if (drawingPath == null || drawingPath.length() == 0) {
             LOG.e(TAG, "Failed to read sketch result from activity");
             this.callbackContext.error("Failed to read sketch result from activity");
             return;
         }
 
         try {
-            String ext = "";
+            if (destinationType == DestinationType.DATA_URL) {
+                String dataenc = "";
+                byte[] drawingData = null;
 
-            if (encodingType == EncodingType.JPEG) {
-                ext = "jpeg";
-            } else if (encodingType == EncodingType.PNG) {
-                ext = "png";
-            }
+                if (encodingType == EncodingType.JPEG) {
+                    dataenc = "jpeg";
+                } else if (encodingType == EncodingType.PNG) {
+                    dataenc = "png";
+                }
```

Comment:
To avoid checking the encoding twice, line 307-311, could the assignment to `dataenc` be moved into that `if` statement